pkg/option/config.go

```diff
@@ -1964,7 +1964,12 @@ func (c *DaemonConfig) populateNodePortRange() error {
 			return errors.New("NodePort range min port must be smaller than max port!")
 		}
 	} else {
-		return errors.New("Unable to parse min/max port value for NodePort range!")
+		// TODO(ctz): This check allows NodePortRange to be passed as explicity
+		// empty, which the pkg/envoy_test.go expects. Should that be the case
+		// or does pkg/envoy_test.go need to be modified?
```

Comment:
After taking another look, I'm going to take a stab at explaining. 

envoy_test.go uses a "preconfigured" `option.Config`: https://github.com/cilium/cilium/blob/640b7a6cb35905ca1aafd3010baf9b8265d383c9/pkg/envoy/envoy_test.go#L68

Then it calls `Populate` which will set additional values in `Config` with values from viper. Due to envoy_test.go running in the test context, the defaults *for* viper haven't been set. They haven't been set because the code which does set defaults is in `daemon_main.go`'s `init()`:

https://github.com/cilium/cilium/blob/5b9002d61500b674c04b5bdd4aa00f91619af8e0/daemon/daemon_main.go#L187

In a non-testing context (running normally), the `init()` mentioned above would run before `Populate` gets called, and everything is OK.

Therefore, in the testing context `Populate` runs standalone meaning that the `init()` mentioned bove does NOT happen before. So it has nothing to retrieve from viper and NodePortRange stays empty.

This is why you can see in the unit tests I wrote (d8f9160), all the pre-test setups are setting up defaults to inform viper, simply because the input to the function under test is viper itself.  

The only reason envoy_test.go complained is because the unit tests I wrote checked for an empty NodePortRange. The code before this PR silently allowed that, whether by accident or on purpose. Either way, it is quite tricky to wrap your head around, but I hope that cleared it up. 

It feels wrong to allow NodePortRange to be empty. I think this all just may have been a coincidental result. We can modify envoy_test.go to reduce its dependency on the entire `DaemonConfig` struct, and only just the options it needs. However I recognize that we may be getting into yak shaving territory here. Let me know what you think.

Reply:
Oh, yeah this seems a bit broken for `pkg/envoy` unit-testing to be relying upon CLI args, but I'm not all that into yak-shaving so it's probably fine to leave it as-is.

There's no big danger here that the guard rails arounnd CLI validation cause inexperienced users to waste lots of time chasing down issues with this, the nodeport range is a pretty advanced configuration and anyone configuring it should likely have an idea about the correct way to pass the values in.